executor/projection.go

```diff
@@ -166,6 +166,8 @@ func (e *ProjectionExec) isUnparallelExec() bool {
 }
 
 func (e *ProjectionExec) unParallelExecute(ctx context.Context, chk *chunk.Chunk) error {
+	// transmit the requiredRows
+	e.childResult.SetRequiredRows(chk.RequiredRows(), e.maxChunkSize)
```

Comment:
`Selection`'s `childResult` needs the same change?

Reply:
Why not? 
It makes the child return as quickly as possible and also guarantees enough rows to return to `Projection's` parent.